server/scripts/populate.py

```diff
@@ -0,0 +1,150 @@
+             "Career", "Engineering", "Science", "Environment", "Arts", "Math", "Health"]
+tag_entities = {}
+
+# Assumptions: each club has its own events. Many-to-many not supported
+# yet. See example clubs.yml file for example input. db.ini file must be
+# in same directory
+
+def insert_tags(): 
+    conn = connect()
+    try:
+        with conn.cursor() as cursor:
+            for tag in tags_enum:
+                _id = str(uuid.uuid4())
+                tag_entities[tag] = _id
+                cursor.execute("INSERT INTO tag VALUES (%s,%s,%s,%s);",
+                               (_id, datetime.now(), datetime.now(), tag))
+        conn.commit()
+        print("Tags Inserted")
+    finally:
+        conn.close()
+
+
+def club_generator(clubs):
+    for club in clubs:
+        yield club
+
+
+def read_clubs(filepath):
+    with open(filepath) as file:
+        clubList = yaml.load(file, Loader=yaml.FullLoader)
+        for club in clubList:
+            if not verify(club, club_non_nullable, "club"):
+                return None
+        return club_generator(clubList)
+
+
+def verify(entity, fields, typeof):
+    for field in fields:
+        if entity.get(field) is None:
+            name = "Untitled" if entity.get("name") is None else entity.get("name")
+            print("Error:", typeof, name,
+                  "is missing mandatory field", field)
+            return False
+        events = entity.get("events")
+        if events is not None:
+            for event in events:
+                if not verify(event, event_non_nullable, "event"):
+                    return False
+        tags = entity.get("tags")
+        if tags is not None:
+            for tag in tags:
+                if tag not in tags_enum:
+                    print("Error:", typeof, entity.get(
+                        "name"), "has illegal tag", tag)
+                    return False
+    return True
+
+
+def insert_clubs():
+    club_list = read_clubs(filepath)
+    if club_list is None:
+        print("Something failed; aborting...")
+        return
+    conn = connect()
+    if conn is None:
+        return
+    insert_tags()
+    try:
+        while True:
+            club = next(club_list)
+            club["id"] = str(uuid.uuid4())
+            # pp.pprint(club)
+            with conn.cursor() as cursor:
+                cursor.execute("INSERT INTO club VALUES (%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s);",
+                               (club.get("id"), datetime.now(), datetime.now(), club.get("name"),
+                                club.get("description"), club.get("size"), club.get(
+                                    "banner_image"), club.get("icon_image"),
+                                club.get("facebook"), club.get("twitter"), club.get("instagram"), club.get("website")))
+                events = club.get("events")
+                if (events is not None):
+                    for event in events:
```

Comment:
Python lets you call `.get` with a default value to return if the key does not exist. This gives us the one-liner

```py
for event in club.get("events", []):
```

If you end up turning clubs into Python objects, then this probably won't be necessary (you can default the field to the empty list, so that it is never `None`), and then `for event in club.events` is enough